nx-X11/programs/Xserver/hw/nxagent/Handlers.c

```diff
@@ -963,12 +963,16 @@ void nxagentHandleCollectInputFocusEvent(int resource)
    * Calling functions from nxcompext from nxagent sounds like a very bad idea
    * to begin with, but let's assume that's necessary for now and work around
    * the corruption issue.
+   *
+   * What we're doing here will lead to compile warnings when XlibWindow is
+   * actually a Window64, but that's "really" the type of "Window" libcompext
+   * expects.
+   *
+   * One way to maybe work around this issue is to define Window, XID and maybe
+   * Atom to their Xlib* types, include NXlib.h and then undefine these macros
+   * again. Might have unforseen consequences, though...
    */
```

Comment:
I am wondering if all this is working correctly in 3.5.0...

Vom Smartphone gesendet.

----- Ursprüngliche Nachricht -----
Von: "Mike Gabriel" <notifications@github.com>
Gesendet: ‎04.‎06.‎2016 00:17
An: "ArcticaProject/nx-libs" <nx-libs@noreply.github.com>
Betreff: Re: [ArcticaProject/nx-libs] nx-X11/programs/Xserver/hw/nxagent/Handlers.h: fix stack smashing rel… (#126)

In nx-X11/programs/Xserver/hw/nxagent/Handlers.c:
> @@ -963,12 +963,16 @@ void nxagentHandleCollectInputFocusEvent(int resource)
>     * Calling functions from nxcompext from nxagent sounds like a very bad idea
>     * to begin with, but let's assume that's necessary for now and work around
>     * the corruption issue.
> +   *
> +   * What we're doing here will lead to compile warnings when XlibWindow is
> +   * actually a Window64, but that's "really" the type of "Window" libcompext
> +   * expects.
> +   *
> +   * One way to maybe work around this issue is to define Window, XID and maybe
> +   * Atom to their Xlib* types, include NXlib.h and then undefine these macros
> +   * again. Might have unforseen consequences, though...
>     */
@Ionic: the workaround you describe is actually the method that gets used in som places of the nxagent code. Please do that, rebase against 3.6.x, so that I can merge this tonight.
Thanks,
Mike
PS: Did you check for other occurrences of #include "NXlib.h" that may also need fixing???
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or mute the thread.